src/screens/WhotoFollow/index.js

```diff
@@ -82,13 +82,33 @@ const WhotoFollow = () => {
     }
   };
 
-  const onNextPage = async (sectionId) => {
-    const checkSectionId = (item) => {
-      return sectionId === 'other'
-        ? item.viewtype === 'labelothers'
-        : item.id === sectionId || item.location_id === sectionId;
+  const getSectionIndex = (sectionId) => {
+    return users?.findIndex((user) => checkSectionId(user, sectionId));
+  };
+
+  const checkSectionId = (item, sectionId) => {
+    return sectionId === 'other'
+      ? item.viewtype === 'labelothers'
+      : item.id === sectionId || item.location_id === sectionId;
+  };
+
+  const getSectionData = (sectionId) => {
+    const data = users?.find((user) => checkSectionId(user, sectionId));
+    return {
+      id: data?.id || data?.location_id || 'others',
+      label: data?.viewtype
     };
-    const indexSectionId = users.findIndex((i) => checkSectionId(i));
+  };
+
+  const onNextPage = async (sectionId) => {
+    const indexSectionId = users.findIndex((i) => checkSectionId(i, sectionId));
+
+    AnalyticsEventTracking.eventTrack(
+      BetterSocialEventTracking.ONBOARDING_WHO_TO_FOLLOW_SEE_MORE_CLICKED,
+      {
+        data: getSectionData(sectionId)
+      }
+    );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `getSectionIndex`, `checkSectionId`, and `getSectionData` functions are introduced to improve the readability of the code. The `onNextPage` function is updated to use these new functions and includes additional analytics tracking. This change enhances the maintainability of the code by breaking down complex logic into smaller, more understandable parts. However, it's important to note that the `users` array is being traversed multiple times within these functions which could impact performance for large data sets. Consider optimizing this part of the code to reduce unnecessary iterations.

```diff
-  const getSectionIndex = (sectionId) => {
-    return users?.findIndex((user) => checkSectionId(user, sectionId));
-  };
-
-  const checkSectionId = (item, sectionId) => {
-    return sectionId === 'other'
-      ? item.viewtype === 'labelothers'
-      : item.id === sectionId || item.location_id === sectionId;
-  };
-
-  const getSectionData = (sectionId) => {
-    const data = users?.find((user) => checkSectionId(user, sectionId));
-    return {
-      id: data?.id || data?.location_id || 'others',
-      label: data?.viewtype
-     };
-  };
-
-  const onNextPage = async (sectionId) => {
-    const indexSectionId = users.findIndex((i) => checkSectionId(i, sectionId));
-
-    AnalyticsEventTracking.eventTrack(
-      BetterSocialEventTracking.ONBOARDING_WHO_TO_FOLLOW_SEE_MORE_CLICKED,
-      {
-        data: getSectionData(sectionId)
-      }
-    );
+  const getSectionData = (sectionId) => {
+    let index = -1;
+    const data = users?.find((user, idx) => {
+      const isMatch = sectionId === 'other'
+        ? user.viewtype === 'labelothers'
+        : user.id === sectionId || user.location_id === sectionId;
+      if (isMatch) index = idx;
+      return isMatch;
+    });
+    return {
+      id: data?.id || data?.location_id || 'others',
+      label: data?.viewtype,
+      index
+    };
+  };
+
+  const onNextPage = async (sectionId) => {
+    const { index, ...data } = getSectionData(sectionId);
+
+    AnalyticsEventTracking.eventTrack(
+      BetterSocialEventTracking.ONBOARDING_WHO_TO_FOLLOW_SEE_MORE_CLICKED,
+      { data }
+    );
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->